src/languages/ruby.js

```diff
@@ -80,6 +80,30 @@ export default function(hljs) {
       }
     ]
   };
+
+  // Ruby syntax is underdocumented, but this grammar seems to be accurate
+  // as of version 2.7.2 (confirmed with (irb and `Ripper.sexp(...)`)
+  // https://docs.ruby-lang.org/en/2.7.0/doc/syntax/literals_rdoc.html#label-Numbers
+  var decimal = '[1-9](_?[0-9])*|0';
+  var digits = '[0-9](_?[0-9])*';
+  var NUMBER = {
+    className: 'number', relevance: 0,
+    variants: [
+      // decimal integer/float, optionally exponential or rational, optionally imaginary
+      { begin: `\\b(${decimal})(\\.(${digits}))?([eE][+-]?(${digits})|r)?i?\\b` },
```

Comment:
Is the r meant to be grouped in with the exponent?

Reply:
Yes that's the "optionally exponential or rational". An implicit decimal literal may be followed by an exponent part _or_ a rational indicator, but not both. `2.718e0` is valid and `2.718r` is valid, but `2.718e0r` is not.